util.py

```diff
@@ -699,10 +723,13 @@ def create_environment(lang, namespace):
     env.globals["param_name"] = language_specific_funcs["param_name"][lang]
     env.globals["escape_keyword"] = language_specific_funcs["escape_keyword"][lang]
     env.globals["custom_type_name"] = language_specific_funcs["custom_type_name"][lang]
+    env.globals["custom_codec_param_name"] = language_specific_funcs["custom_codec_param_name"][lang]
     env.globals["get_size"] = get_size
     env.globals["is_trivial"] = is_trivial
-    env.globals["get_import_path_holders"] = language_specific_funcs["get_import_path_holders"][
-        lang
-    ]
+    env.globals["get_import_path_holders"] = language_specific_funcs["get_import_path_holders"][lang]
+    env.globals["copyright_year"] = date.today().year
+    env.globals["protocol_commit"] = os.popen("git rev-parse --short HEAD").readlines()[0].strip()
```

Comment:
I think this wouldn't work if this repo was downloaded (instead of cloned), or `git` is not available in the `PATH`. Should we set`protocol_commit` to some default if an exception is thrown here?

Reply:
The correct usage of `os.poopen` would be:
```
with os.popen("git rev-parse --short HEAD") as f:
  env.globals["protocol_commit"] = f.readlines()[0].strip()
```

Reply:
can do, pushed a fix